proto/sdk/alpha/alpha.proto

```diff
@@ -20,4 +20,56 @@ option go_package = "alpha";
 import "google/api/annotations.proto";
 
 // SDK service to be used in the GameServer SDK to the Pod Sidecar
-service SDK {}
\ No newline at end of file
+service SDK {
+    // Call when a player has connected
+    rpc PlayerConnect (PlayerId) returns (Empty) {
+        option (google.api.http) = {
+            post: "/alpha/player/connect"
+            body: "*"
+        };
+    }
+
+    // Call when a player has disconnected
+    rpc PlayerDisconnect (PlayerId) returns (Empty) {
+        option (google.api.http) = {
+            post: "/alpha/player/disconnect"
+            body: "*"
+        };
+    }
+
+    // change the player capacity to a new value
+    rpc SetPlayerCapacity (Count) returns (Empty) {
+        option (google.api.http) = {
+            post: "/alpha/player/capacity"
+            body: "*"
+        };
+    }
+
+    // get the current player capacity
+    rpc GetPlayerCapacity(Empty) returns (Count) {
+        option (google.api.http) = {
+            get: "/alpha/player/capacity"
+        };
+    }
+
+    // get the current player count
+    rpc GetPlayerCount(Empty) returns (Count) {
+        option (google.api.http) = {
+            get: "/alpha/player/count"
+        };
+    }
+}
+
+// I am Empty
+message Empty {
+}
+
+// Store a count variable
+message Count {
+    int64 count = 1;
+}
+
+// The unique identifier for a given player
+message PlayerId {
```

Comment:
I'm wondering if a single string is sufficient for this. What if you have multiple types of "players"? With a single string field the programmer will end up doing some funny string concatenation / parsing logic to shove it all into one field. If we had the option for some generic metadata (labels?), then you could use that instead to namespace different types of players.

I'm not necessarily saying we should change it right now (so we don't fall into the trap of over-designing), but I'm wondering if there is a way for us to get some feedback from folks that would actually use this. Maybe the best way is to get it in as alpha so folks can start playing around with it?

Reply:
So this is the design that was outlined [in the accompanying ticket ](#1033), and nobody disagreed with it. (Also, in 1:1 discussions, everyone seemed okay with the design).

_puts opinionated hat on_ :tophat: I don't think it's Agone's job to be storing/transferring data about players - that's why we only use a playerId. Any other metadata about players a game system wants to store, they should store that separately, and attach it in that system to this unique id that is passed between Agones and other systems for identification purposes.

Reply:
Seems reasonable. It sounds like you've thought about this and gotten some feedback.